### PR TITLE
Add ai

### PR DESCRIPTION
**Pull request by agamm**
- Adds ai to generate project changes from snippets.
- Added metadata to snippets and relevant function to load them
- Added openai key in .env
- Added prompts file with prompts to generate our outputs.
- Rewrote the bun cli file to use commander.

https://www.loom.com/share/5eb5f6ef39c34a7f86007d04b4aa8f08?sid=7c9776aa-3534-4571-b09d-00259b5c8bdb

### CODE REVIEW

File: aleph0/prompts.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+export const findRelevantSnippets = (
+  userText: string,
+  snippetsText: string
+) => `
+Given the following snippets, 
+in the following format per snippet: '{name} - {description} (path: {path})\n'
+${snippetsText}
+
+Output should be a valid json from abovesnippets.
```
Comment (elie222) [inline aleph0/prompts.ts]: *above snippets
Or "from the snippets above"

File: aleph0/prompts.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  projectStructure: string
+) => {
+  const snippetObj = JSON.parse(snippet);
+  const mappedSnipepts = (await getSnippetFiles(snippetObj.path)).join("\n");
+
+  return `
+  You are an expert Next13 fullstack developer.
+  In version 13, Next.js introduced a new App Router built on React Server Components, which supports shared layouts, nested routing, loading states, error handling, and more1
+3
+. The App Router works in a new directory named app, which allows for greater flexibility in terms of project structure3
```
Comment (elie222) [inline aleph0/prompts.ts]: structure3 => structure3

File: aleph0/prompts.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+. The new App Router structure does not include a pages/ directory, which was used in the previous version2
```
Comment (elie222) [inline aleph0/prompts.ts]: version2 => version

Comment (elie222) [reply aleph0/prompts.ts]: it can still use pages dir btw (many projects still use both)

File: aleph0/prompts.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  ${mappedSnipepts}
+
+  Rules:
+  - Valid JSON array, no explanations.
+  - If the file needs to be created, think about the most apropriate path and file name.
+    - Add an attribute {create: true} if so.
+  - snippet and sourcePath have to have the same file name (e.g. snippet="route.ts", then sourcePath must be a file named the same: "route.ts")
+
+  Create a JSON array in the following format example:
+  [{snippet: "snippets/search/route.ts", sourcePath: "app/search/route.ts"]
```
Comment (elie222) [inline aleph0/prompts.ts]: function call another approach to get json back but if this works well then leave as is

File: aleph0/prompts.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    Folders are used to define routes.
+    Each folder in a route represents a route segment.
+    Each route segment is mapped to a corresponding segment in a URL path.
+    The new App Router structure does not include a pages/ directory.
+
+  Assuming the following project structure:
+  ${projectStructure}
+
+  And the following snippet files:
+  ${mappedSnipepts}
```
Comment (elie222) [inline aleph0/prompts.ts]: typo `mappedSnipepts`

File: aleph0/index.ts [hunk 6]
```diff
@@ -1 +1,87 @@
...
+const program = new Command();
+program
+  .command("gen <text>")
+  .description("Generate nextjs snippet.")
+  .action((text) => {
+    generate(text);
+  });
+program.parse(process.argv);
+
+// example to create an "article" component with an api route
```
Comment (elie222) [inline aleph0/index.ts:68]: feel free to delete this code when you want btw

Comment (agamm) [reply aleph0/index.ts:68]: Not yet.

File: aleph0/prompts.ts (hunk 4 above)
Comment (agamm) [reply aleph0/prompts.ts]: Right now it seems to work :) I tried KISS like you said

**Review by elie222 — APPROVED** [review]

**Pull request merged**